geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/CreateMissingBucketsTask.java

```diff
@@ -58,4 +70,38 @@ protected void createMissingBuckets(PartitionedRegion region) {
       }
     }
   }
+
+  /**
+   * Wait for Colocation to complete. Wait all nodes to Register this PartitionedRegion.
+   */
+  private boolean waitForColocationCompleted(PartitionedRegion partitionedRegion) {
+    int retryCount = 0;
+    int sleepInterval = PartitionedRegionHelper.DEFAULT_WAIT_PER_RETRY_ITERATION;
+
+    while (!ColocationHelper.isColocationComplete(partitionedRegion)
```

Comment:
`ColocationHelper.isColocationComplete` will need to wrapped in a `java.util.function.Function` as well to control this for unit testing.